### DISCUSSION TITLE
History Log ( prompt log)

### DISCUSSION THREAD

**Discussion by K3nt3L** [lllyasviel/Fooocus#2012] [Ideas]
Hi,

After using Foocus extensively, I have a suggestion to improve the user experience and make it easier for users to recreate the same or similar images, either with Foocus or another software, like ComfyUI, for example ;).

There's a crucial missing feature in the software: the ability to see not just the entered prompt (and any potential tokens from Foocus V2) and the selected styles, but the actual instruction sent to condition the sampling.

Additionally, when selecting multiple styles, it becomes unclear in what order the tokens are actually written. It would be great if you could add a line to the log for this information.

Thanks!

Example
Now :
Prompt | Pulp comics movie poster, An antropomorphic monkey wearing a astronaut suit and metallic mittens, action, in an bioluminescent airlock, MoviePosterAF
-- | --
Negative Prompt | rodent, extra limbs, extra fingers, extra legs
Fooocus V2 Expansion | Pulp comics movie poster, An antropomorphic monkey wearing a astronaut suit and metallic mittens, action, in an bioluminescent airlock, MoviePosterAF, magical, intricate, elegant, highly detailed, extremely generous, inspired, innocent, loving, color, dramatic light, epic, fantastic, cinematic, fine detail, pretty, perfect, colorful
Styles | ['Fooocus V2', 'Mk Atompunk']

You can add the real prompt  :
Atompunk illustation,  Pulp comics movie poster, An antropomorphic monkey wearing a astronaut suit and metallic mittens, action, in an bioluminescent airlock, MoviePosterAF, magical, intricate, elegant, highly detailed, extremely generous, inspired, innocent, loving, color, dramatic light, epic, fantastic, cinematic, fine detail, pretty, perfect, colorful, retro-futuristic, atomic age aesthetics, sleek lines, metallic textures, futuristic technology, optimism, energy
Negative : rodent, extra limbs, extra fingers, extra legs, organic, natural textures, rustic, dystopian.

Note : And I'm not sure it's the exact prompt...

Hope you can help. 
Thanks for all the hard job.

**Comment by poor7** [comment]
I would also like to know the exact clues and their relative positions. You can also make an optional switch in the advanced menu.

**Comment by mashb1t** [comment]
None of the styles prompts (positive/negative) are included in Fooocus V2 prompt expansion output.
Please find the code of how the prompt is built in https://github.com/lllyasviel/Fooocus/blob/main/modules/async_worker.py#L383 and below.
I'm currently working on recreation of images using metadata, check it out in PR https://github.com/lllyasviel/Fooocus/pull/1940
The import is locally WIP, not yet pushed.

**Reply by K3nt3L** [reply]
It's why I asking for... Not only  wrtiting user's prompt and style's names in the log, but complete real prompt (and V2 in case of use) 

The  no-metadata picture seems a clear choice for the dev. 
No pb with that, I could find some good reasons for this too.

Even if the final conditionning and sampling is different  from Comfy or Automatic1111, it would be useful for Foocus's users (educational and consistency). It's not about recreating the same picture with another UI. Just a better understanding of what happened in the black box.

**Comment by docppp** [comment]
https://github.com/lllyasviel/Fooocus/pull/1920

this PR adds positive and negative full prompt like this:

![obraz](https://github.com/lllyasviel/Fooocus/assets/29142757/c3513ed1-4829-492a-9af8-a93398605c6d)

**Reply by eddyizm** [reply]
@docppp That is a really neat feature. I may pull that change into my fork. I was looking at trying to get image prompt info too but did not have any luck looking for it the other night.
